tests/pipelines/test_pipelines_common.py

```diff
@@ -1041,6 +1044,218 @@ def _test_attention_slicing_forward_pass(
+        original_pipe_components = {}
+        # additional components that are not in the pipeline, but expected in the original pipeline
+        original_pipe_additional_components = {}
+        # additional components that are in the pipeline, but not expected in the original pipeline
+        additional_components = {}
+        # components expected in the original pipeline
+        original_expected_modules, _ = original_pipeline_class._get_signature_keys(original_pipeline_class)
+
+        for k, v in components.items():
+            if k in original_expected_modules:
+                original_pipe_components[k] = v
+            else:
+                additional_components[k] = v
+        for k in original_expected_modules:
+            if k not in original_pipe_components:
+                if k in original_pipeline_class._optional_components:
+                    original_pipe_additional_components[k] = None
+                else:
+                    raise ValueError(f"missing expected module {k}")
+
+        # original pipeline: (sd/sdxl)
+        pipe_original = original_pipeline_class(**original_pipe_components, **original_pipe_additional_components)
+        pipe_original.to(torch_device)
+        pipe_original.set_progress_bar_config(disable=None)
+
+        original_config = {k: v for k, v in pipe_original.config.items() if not k.startswith("_")}
+
+        inputs = get_dummy_inputs_pipe_original(torch_device)
+        output_original = pipe_original(**inputs)[0]
+
+        # pipe1: pipeline
+        pipe1 = self.pipeline_class(**components)
+        pipe1.to(torch_device)
+        pipe1.set_progress_bar_config(disable=None)
+        pipe1_config = {k: v for k, v in pipe1.config.items() if not k.startswith("_")}
+
+        inputs = get_dummy_inputs(torch_device)
+        output1 = pipe1(**inputs)[0]
+
+        # pipe2 (created with `from_pipe`): original pipeline(sd/sdxl) -> pipeline
+        pipe2 = self.pipeline_class.from_pipe(pipe_original, **additional_components)
+        pipe2.to(torch_device)
+        pipe2.set_progress_bar_config(disable=None)
+        pipe2_config = {k: v for k, v in pipe2.config.items() if not k.startswith("_")}
+
+        inputs = get_dummy_inputs(torch_device)
+        output2 = pipe2(**inputs)[0]
+
+        # pipe_original_2 (created with `from_pipe`): pipeline ->  original pipeline (sd/sdxl)
+
+        ## for pipelines such as animtediff and pia, the unet changes from UNet2DConditionModel to UNetMotionModel during __init__
+        changed_components = {}
+        original_signature_types = original_pipeline_class._get_signature_types()
+        for k, v in pipe2.components.items():
+            if (
+                k in original_expected_modules
+                and v is not None
+                and isinstance(v, torch.nn.Module)
+                and type(v) not in original_signature_types[k]
+            ):
+                changed_components[k] = original_pipe_components[k]
```

Comment:
I am really having trouble reading this condition. 

* Why should the `original_pipe_additional_components` go into initializing a different pipeline that might not share the same additional components?

Reply:
Maybe fully spelling out what `k` and `v` are denoting will be helpful.

Reply:
> Why should the original_pipe_additional_components go into initializing a different pipeline that might not share the same additional components?

yeah, I know - not the most straight-forward code to read here 
so we only have two pipelines here, the original pipeline would be something like SD, the current pipeline would be something like SAG

Basically in the test we do something like this, 
                  
                        SD (`pipe_original`) -> SAG (`pipe2`) -> SD (`pipe_original_2`)

so you can see that `pipe_original_2` and pipe_original` is actually same pipeline class, `original_pipe_additional_components` would be components in `pipe_original` that not accepted in `pipe2`, e.g. `image_encoder` (I'm just making it up here), naturally it will go back when we initiate the `pipe_original_2` since it is the same pipeline class 

maybe I should give them better names!